### DISCUSSION TITLE
Golang nonFixed tests

### DISCUSSION THREAD

**Discussion by AharonSambol** [codewars/codewars.com#2443] [Q&A] [answered]
Hi I'm trying to write a kata in golang but my puzzles could have more than one solution each, I can't seem to figure out how to test this...
(I have a function which can test whether or not a solution is valid)
thanks in advance!

**Comment by kazk** [comment, marked as answer]
Do you mean something equivalent to `assert(test(Solve(puzzle)))`? `test` returns `true` if the solution is valid. I think it's `Expect(test(Solve(puzzle))).To(Equal(true))` in Gomega.

You can also implement your own custom assertion. See https://onsi.github.io/gomega/

If you can describe what you're having trouble with more specifically, it's easier to help.

**Reply by AharonSambol** [reply]
Yup that looks like it should be it 
Thanks for the help I'll try and get it to work
